### ISSUE TITLE
Windows error on loading ../magics/shell_magic.py

### ISSUE THREAD

**Issue by fbcotter** [Calysto/metakernel#94]
I downloaded matlab kernel which I think uses the metakernel template.
When I tried to start a new notebook I was getting the error:

"Can't load PATH_TO_METAKERNEL\magics\shell_magic.py: error: The command was not found or was not executable: sh.

I had a look at the code and think I see the problem:
```python
def start_process(self):
 ...
     if not self.cmd:
         try:
             pexpect.which('bash')
         except Exception as e:  # pragma: no cover
             if os.name == 'nt':
                 self.cmd = 'cmd'
                 self.repl = cmd()
            else:
                raise(e)
        else:
            if pexpect.which('bash'):
                self.cmd = 'bash'
                self.repl = bash()
            else:
                self.cmd = 'sh'
                self.repl = bash(command='sh')
```
The `pexpect.which('bash')` doesn't throw an error if it couldn't find bash, but just returns none, so it continues down the Unix path and sets the command to 'sh'. 
I changed the code in my local build to force it into the windows path and it works now, but I don't wan't to do a pull request as I don't quite understand what's happening in the Unix path in choosing between sh and bash.

**Comment by dsblank** [comment]
Ok, thanks. Can you post your changes that do work? That can help us piece together a solution.

**Comment by fbcotter** [comment]
Oops, sorry for slow reply - I'd set up the wrong email set up to get notifications. 

For the windows section (choosing cmd instead of bash), this worked:
```python
def start_process(self):
    if not self.repl is None:
        self.repl.child.terminate()

    if not self.cmd:
        if not pexpect.which('bash'):
            if os.name == 'nt':
                self.cmd = 'cmd'
                self.repl = cmd()
            else:
                raise Exception("Error: The command was not found or was not executable: sh")           
        else:
            if pexpect.which('bash'):
                self.cmd = 'bash'
                self.repl = bash()
            else:
                self.cmd = 'sh'
                self.repl = bash(command='sh')

```

**Comment by blink1073** [comment]
Should be fixed by 0c6d9a1e841d83e5f6833c2c2c72dabd9773dadb, released in `0.12.5`.
